frappe/website/path_resolver.py

```diff
@@ -105,7 +105,9 @@ def resolve_redirect(path, query_string=None):
 	                ]
 	"""
 	redirects = frappe.get_hooks("website_redirects")
-	redirects += frappe.get_all("Website Route Redirect", ["source", "target"], order_by=None)
+	redirects += frappe.get_all(
+		"Website Route Redirect", ["source", "target", "redirect_http_status"], order_by=None
+	)
 
 	if not redirects:
 		return
```

Comment:
Suggestion: Also cache the status code

```python
	redirect_to = frappe.cache.hget("website_redirects", path)

	if redirect_to:
		redirect_to, redirect_http_status = redirect_to  # QUESTION: Maybe this breaks if cache is stale and contains non-tuple values? Should we wrap it with a `if isinstance(..., tuple)`?
		frappe.flags.redirect_location = redirect_to
		raise frappe.Redirect(redirect_http_status or 301)
```

_GitHub's UI doesn't allow commenting on non-modified lines for some reason._